src/Limiters/limiter.jl

```diff
@@ -0,0 +1,115 @@
+        Qstar_e_mass[e] = sum(Fields.slab(Qstar, e))
+
+        # This should never happen, but if it does, don't limit
+        if ρstar_e_mass[e] <= 0
+            continue
+        end
+
+        # Relax constraints to ensure limiter has a solution:
+        # This is only needed if runnign with the SSP CFL>1 or
+        # due to roundoff errors.
+        if Qstar_e_mass[e] < min_Q[e] * ρstar_e_mass[e]
+            min_Q[e] = Qstar_e_mass[e] / ρstar_e_mass[e]
+        end
+
+        if Qstar_e_mass[e] > max_Q[e] * ρstar_e_mass[e]
+            max_Q[e] = Qstar_e_mass[e] / ρstar_e_mass[e]
+        end
+
+        local_geometry_slab = Fields.slab(space.local_geometry, e)
+
+        for iter in 1:(Nq * Nq - 1)
+            mass_change = 0.0
+            # Iterate over quadrature points
+            for j in 1:Nq, i in 1:Nq
+                ρstar_wJ[i, j] =
+                    local_geometry_slab[i, j].WJ * parent(ρstar)[i, j, 1, e]
+                # Compute the error tolerance and project qstar into the
+                # upper and lower bounds
+                if qstar[i, j, 1, e] > max_Q[e]
+                    mass_change +=
+                        (qstar[i, j, 1, e] - max_Q[e]) * ρstar_wJ[i, j]
+                    qstar[i, j, 1, e] = max_Q[e]
+                elseif qstar[i, j, 1, e] < min_Q[e]
+                    mass_change -=
+                        (min_Q[e] - qstar[i, j, 1, e]) * ρstar_wJ[i, j]
+                    qstar[i, j, 1, e] = min_Q[e]
+                end
+            end
+
+            # By this projection of qstar to the upper/lower bounds, we
+            # have introduced a mass_change. Check if we are within the chosen
+            # tolerance
+            if abs(mass_change) <= tol * abs(Qstar_e_mass[e])
+                break
+            end
+
+            weights_sum = 0.0
+
+            # If the change was positive, the removed mass is added
+            if mass_change > 0
+                # Iterate over quadrature points
+                for j in 1:Nq, i in 1:Nq
+                    if qstar[i, j, 1, e] < max_Q[e]
+                        weights_sum += ρstar_wJ[i, j]
+                    end
+                end
+                for j in 1:Nq, i in 1:Nq
+                    if qstar[i, j, 1, e] < max_Q[e]
+                        qstar[i, j, 1, e] += mass_change / weights_sum
+                    end
+                end
+            else # If the change was negative, the added mass is removed
+                # Iterate over quadrature points
+                for j in 1:Nq, i in 1:Nq
+                    if qstar[i, j, 1, e] > min_Q[e]
+                        weights_sum += ρstar_wJ[i, j]
+                    end
+                end
+                for j in 1:Nq, i in 1:Nq
+                    if qstar[i, j, 1, e] > min_Q[e]
+                        qstar[i, j, 1, e] += mass_change / weights_sum
+                    end
+                end
+            end
+        end
+    end
```

Comment:
We can update `Qstar` inside the loop
```suggestion
        Fields.slab(Qstar, e) .= qstar .* Fields.slab(ρstar, e)
    end
```

Reply:
@simonbyrne I tried this suggestion (with the new variable names, i.e., removing the "star" part), but I see this error never encountered before on this line:
```
ArgumentError: Broadcasted{Unknown} wrappers do not have a style assigned
```